pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -386,6 +387,39 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (reconcile
 		}
 	}
 
+	// Check the current power status against the desired power
+	// status.
+	if provResult, err = prov.UpdateHardwareState(); err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "failed to update the hardware status")
+	}
+	if provResult.Dirty {
+		if err := r.saveStatus(host); err != nil {
+			return reconcile.Result{}, errors.Wrap(err,
+				"failed to update host hardware state")
+		}
+		return reconcile.Result{RequeueAfter: provResult.RequeueAfter}, nil
+	}
+	if host.Status.PoweredOn != host.Spec.Online {
+		reqLogger.Info("power state change needed",
+			"expected", host.Spec.Online, "actual", host.Status.PoweredOn)
+		if host.Spec.Online {
+			provResult, err = prov.PowerOn()
+		} else {
+			provResult, err = prov.PowerOff()
+		}
+		if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "failed to manage power state of host")
+		}
+		if provResult.Dirty {
+			if err := r.saveStatus(host); err != nil {
+				return reconcile.Result{}, errors.Wrap(err,
+					"failed to save host status after power change")
+			}
+			return reconcile.Result{Requeue: true, RequeueAfter: provResult.RequeueAfter}, nil
```

Comment:
It looks like the `Requeue` field is [ignored](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.1.10/pkg/internal/controller/controller.go#L221-L229) if `RequeueAfter` is set.

Reply:
Depending on the state of things, the delay won't always be set by the provisioner, so explicitly setting Requeue to true here ensures that we're requeued.